doc/7/getting-started/.react/standalone/src/App.js

```diff
@@ -0,0 +1,155 @@
+/* snippet:start:1 */
+import React from 'react';
+import kuzzle from "./services/kuzzle";
+import './App.css';
+/* snippet:end */
+/* snippet:start:8 */
+const Message = function (props) {
+  const { message, username } = props;
+  return (
+    <div className={(message.username === username ? 'fromMe' : 'fromOthers') + ' messages'}>
+      <span> User: <b>{message.username}</b>  (</span>
+      <span> {new Date(message.createdAt).toLocaleString().split("GMT")[0]} )</span>
+      <p> {message.value} </p>
+    </div>
+  );
+}
+/* snippet:end */
+/* snippet:start:2 */
+class App extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      username: "",
+      message: "",
+      messages: [],
+      validate: false,
+    };
+  }
+/* snippet:end */
+  /* snippet:start:3 */
+  valid = async () => {
+    await kuzzle.connect();
+    if (!await kuzzle.index.exists("chat")) {
+      await kuzzle.index.create("chat");
+      await kuzzle.collection.create("chat", "messages");
+    }
+    await this.fetchMessages();
+    await this.subscribeMessages();
+    this.setState({ validate: true });
+  }
+  /* snippet:end */
+  /* snippet:start:10 */
+  subscribeMessages = async () => {
+    await kuzzle.realtime.subscribe(
+      "chat",
+      "messages", {},
+      notification => {
+        if (notification.type !== "document") return;
+        if (notification.action !== "create") return;
+        this.setState({
+          messages: [
+            this.getMessage(notification.result),
+            ...this.state.messages
+          ]
+        })
+      }
+    );
+  }
+  /* snippet:end */
+  /* snippet:start:6 */
+  getMessage = (document) => {
+    const message = {
+      _id: document._id,
+      value: document._source.value,
+      createdAt: document._source._kuzzle_info.createdAt,
+      username: document._source.username
+    };
+    return message;
+  }
+  /* snippet:end */
+  /* snippet:start:7 */
+fetchMessages = async () => {
+    const results = await kuzzle.document.search(
+      "chat",
+      "messages", { sort: ["_kuzzle_info.createdAt"] }, { size: 100 }
+    );
+    results.hits.map(hit => {
+      this.setState({ messages: [this.getMessage(hit), ...this.state.messages] })
+    });
+  }
+  /* snippet:end */
+  /* snippet:start:5 */
+  handleInputChange = (event) => {
```

Comment:
Nitpicking: avoid unnecessary parenthesis
```suggestion
  handleInputChange = event => {
```

Also I'm not comfortable with this syntax instead of
```
function handleInputChange() 
```

Is it a standard in React to use this syntax?

Reply:
I'm using arrow functions by abitude. In this case with the `handleInputChange` function it is to avoid having to `bind` `this`